lib/promoted/ruby/client/util.rb

```diff
@@ -3,7 +3,11 @@ module Ruby
       module Client
         module Util
             def self.translate_args(args)
-              args.transform_keys { |key| key.to_s.to_underscore.to_sym }
+              sym_hash = {}
+              args.each do |k, v|
+                sym_hash[k.to_s.to_underscore.to_sym] = v.is_a?(Hash) ? translate_args(v) : v
```

Comment:
Longer-term, I'd be curious what the performance impact of this.  E.g. are all these strings cached?  Or are new strings generated for each to_underscore?

Reply:
Not worth doing in this PR.

Reply:
Mostly this is going to be cost of doing business. One potential improvement here is if the key is already a symbol, don't run it back and forth through the transformations. I can investigate doing that.